libs/libskynet/src/checkObj.ts

```diff
@@ -0,0 +1,35 @@
+import { Err } from "./types.js";
+
+// checkObj take an untrusted object and a list of typechecks to perform and
+// will check that the object adheres to the typechecks. If a type is missing
+// or has the wrong type, an error will be returned. This is intended to be
+// used to check untrusted objects after they get decoded from JSON. This is
+// particularly useful when receiving objects from untrusted entities over the
+// network or over postMessage.
+//
+// Below is an example object, followed by the call that you would make to
+// checkObj to verify the object.
+//
+// const expectedObj = {
+//   aNum: 35,
+//   aStr: "hi",
+//   aBig: 10n,
+// };
+//
+// const err = checkObj(expectedObj, [
+//   ["aNum", "number"],
+//   ["aStr", "string"],
+//   ["aBig", "bigint"],
+// ]);
+function checkObj(obj: any, checks: [string, string][]): Err {
+  for (let i = 0; i < checks.length; i++) {
+    const check = checks[i];
+    const type = typeof obj[check[0]];
```

Comment:
should we add a special case for "array" ?

Reply:
Yes since we're probably be going to be expecting arrays at some point - added in https://github.com/SkynetLabs/skynet-kernel/pull/253/commits/5e870460fcb74a0572c8bf8229b03cacbe43332d